docker-rke2.yaml

```
@@ -156,7 +156,7 @@ data:
       http-check expect status 401
       # TODO: we should be verifying (!)
       {{range $server, $address := .BackendServers}}
-      server {{ $server }} {{ JoinHostPort $address $.BackendControlPlanePort }} check check-ssl verify none resolvers docker resolve-prefer {{ if $.IPv6 -}} ipv6 {{- else -}} ipv4 {{- end }}
+      server {{ $server }} {{ JoinHostPort $address.Address $.BackendControlPlanePort }} check check-ssl verify none resolvers docker resolve-prefer {{ if $.IPv6 -}} ipv6 {{- else -}} ipv4 {{- end }}
```

nice finding!

---

```suggestion
      {{range $server, $backend := .BackendServers }}
      server {{ $server }} {{ JoinHostPort $backend.Address $.BackendControlPlanePort }} check check-ssl verify none resolvers docker resolve-prefer {{ if $.IPv6 -}} ipv6 {{- else -}} ipv4 {{- end }}
      {{- end}}
```
to align with the template in the turtles repo: https://github.com/rancher/turtles/blob/85e718b7ef4d78cf6273dab603dda0fac9527c24/test/e2e/data/cluster-templates/docker-rke2.yaml

Maybe also it needs to be mentioned that this change works with CAPI 1.9+ only